### PR TITLE
NXP-25257: decrypt variables in nxftl context

### PR DESCRIPTION
**Pull request by pierre-gautier**

### CODE REVIEW

**Comment by kevinleturc** [comment]
👍

File: nuxeo-common/src/main/java/org/nuxeo/common/utils/TextTemplate.java [hunk 1]
```diff
@@ -261,7 +261,7 @@ public Properties preprocessVars(Properties unprocessedVars) {
             }
             recursionLevel++;
             // Avoid infinite replacement loops
-            if ((!doneProcessing) && (recursionLevel > MAX_RECURSION_LEVEL)) {
+            if (!doneProcessing && recursionLevel > MAX_RECURSION_LEVEL) {
```
Comment (efge) [inline nuxeo-common/src/main/java/org/nuxeo/common/utils/TextTemplate.java]: Please keep cleanups in separate commits (i.e., don't have your IDE automatically do cleanups).

File: nuxeo-common/src/main/java/org/nuxeo/common/utils/TextTemplate.java [hunk 2]
```diff
@@ -324,10 +324,14 @@ public void initFreeMarker() {
         String currentString;
         KEYS: for (String key : processedVars.stringPropertyNames()) {
             String value = processedVars.getProperty(key);
+            if (value.startsWith("${") && value.endsWith("}")) {
+                // crypted variables have to be decrypted in freemarker vars
```
Comment (efge) [inline nuxeo-common/src/main/java/org/nuxeo/common/utils/TextTemplate.java]: crypted -> encrypted

**Comment by efge** [comment]
Could you please add a unit test.

**Comment by efge** [comment]
👍
